core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerImpl.java

```diff
@@ -405,6 +405,13 @@ void create(int tableId, TableStructure tableStruct) {
             metadata.create(tableStruct, tableToken, path, rootLen, tableId);
             tableTransactionLog.create(path, timestamp);
             engine.getWalListener().tableCreated(tableToken, timestamp);
+        } catch (Throwable th) {
```

Comment:
considering this method is used from one place, it would be simpler (less spaghetti) if `create` functionality was called from the constructor. This way the class will behave just like everything else.

Reply:
I didn't understand this comment, we should elaborate more.

Reply:
There are two constructor calls, and only one of them is followed by `create()`. Inlining `create()` into the constructor would change the behaviro.